static/search.js

```diff
@@ -0,0 +1,72 @@
+function debounce(func, timeout = 200) {
+    let timer;
+    return (...args) => {
+        clearTimeout(timer);
+        timer = setTimeout(() => {
+            func.apply(this, args);
+        }, timeout);
+    };
+}
+
+let boxUpper = document.getElementById('searchBoxUpper');
+let boxLower = document.getElementById('searchBoxLower');
```

Comment:
could just do something like
```suggestion
const searchBox = document.getElementById('searchBoxUpper') || document.getElementById('searchBoxLower');
```

that way you don't have to check which one is non-null since at least one of them will be rendered at any given point

Reply:
did it the first way to account for screen resizing, but have now realized close to nobody will be doing that.